File: Low Level Design (LLD)/DesignPatterns/3. Structural/Interpreter/Java/Context.java
import java.util.HashMap;
import java.util.Map;

public class Context {
    private Map<String, Integer> variableMap = new HashMap<>();

    public void assign(String variable, int value) {
        variableMap.put(variable, value);
    }

    public int getValue(String variable) {
        if (!variableMap.containsKey(variable)) {
            throw new IllegalArgumentException("Variable not defined: " + variable);
        }
        return variableMap.get(variable);
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Interpreter/Cpp/src/Context.cpp
#include "../include/Context.h"

void Context::assign(const std::string& var, int value) {
    variables[var] = value;
}

int Context::lookup(const std::string& var) const {
    auto it = variables.find(var);
    if (it != variables.end()) return it->second;
    return 0;
}
